cmd/aks-periscope/aks-periscope.go

```diff
@@ -41,10 +42,19 @@ func main() {
 	collectors = append(collectors, kubeletCmdCollector)
 	systemPerfCollector := collector.NewSystemPerfCollector(exporter)
 	collectors = append(collectors, systemPerfCollector)
-	osmLogsCollector := collector.NewOsmLogsCollector(exporter)
-	collectors = append(collectors, osmLogsCollector)
-	smiLogsCollector := collector.NewSmiLogsCollector(exporter)
-	collectors = append(collectors, smiLogsCollector)
+
+	// Checks flags and adds specified collector if matched
+	flags := strings.Fields(os.Getenv("FLAGS_LIST"))
+	for _, flag := range flags {
+		if flag == "OSM" {
```

Comment:
would it make sense to use a `switch case` here so that it exits on the first match?

Reply:
> Could you give an example of how the flag is used? i.e., have you tested it with the az/kubectl CLI or does this change expect the value to be set in the yaml?

actually spoke to someone on the periscope team and it seems like this is going to be used via the yaml only. Feature flags are going to be part of long term refactoring work